pkg/ipam-controller/allocator/allocator.go

```diff
@@ -161,8 +165,8 @@ func (pa *poolAllocator) checkAllocation(allocData nodeAllocationInfo) error {
 	// check that StartIP of the range has valid offset.
 	// all ranges have same size, so we can simply check that (StartIP offset - 1) % pa.cfg.PerNodeBlockSize == 0
 	// -1 required because we skip network addressee (e.g. in 192.168.0.0/24, first allocation will be 192.168.0.1)
-	if math.Mod(float64(ip.Distance(ip.Network(pa.cfg.Subnet).IP, allocData.StartIP))-1,
-		float64(pa.cfg.PerNodeBlockSize)) != 0 {
+	distanceFromGW := ip.Distance(pa.cfg.Subnet.IP, allocData.StartIP)
```

Comment:
distanceFromNetworkStart ? 

GW may not bet the .1 address in the network.

Reply:
also see my comment regarding dropping of ip.Network() call

Reply:
Yes, the variable name is incorrect. I will rename it

Reply:
I renamed the variable